### PR TITLE
Implements Application Model Save and Delete

### PR DESCRIPTION
**Pull request by mccraveiro**

### CODE REVIEW

File: src/Model.js [hunk 1]
```diff
@@ -34,29 +40,42 @@ export default class Model {
...
 
-  [exec_middlewares](timing, action) {
+  [execute_middleware](timing, action) {
     this[middlewares][timing][action].forEach(middleware => {
       middleware.call(this)
     })
   }
 
+  static get_collection() {
+    let database = monk('localhost/app')
```
Comment (brenolf) [inline src/Model.js]: This path shouldn't be hardcoded here, the user must be able to choose which db to use

**Comment by mccraveiro** [comment]
@brenolf please review

File: src/Cheddar.js [hunk 2]
```diff
@@ -1,3 +1,13 @@
...
+
+export class Cheddar {
+  static get database() {
+    return this._database
+  }
+
+  static set database(url) {
+    this._database = url
+  }
+}
```
Comment (brenolf) [inline src/Cheddar.js]: spec?

Comment (mccraveiro) [inline src/Cheddar.js]: I won't add a test for a simple getter and setter. If it gets more complex we can add tests later.

File: src/Model.js [hunk 3]
```diff
@@ -9,9 +15,10 @@ export default class Model {
     }
 
     this.configure && this.configure()
-    this[exec_middlewares]('before', 'create')
-    // TODO: create model
-    this[exec_middlewares]('after', 'create')
+    this[execute_middleware]('before', 'create')
+    this[collection] = this.constructor.collection
+    this.id = this[collection].id()
```
Comment (brenolf) [inline src/Model.js]: What's `id`?

Comment (mccraveiro) [inline src/Model.js]: The document ID on the database. http://docs.mongodb.org/manual/reference/object-id/

File: src/Cheddar.js (hunk 2 above)
Comment (brenolf) [inline src/Cheddar.js]: Test it, even if its simple. Save time for later work :wink:

File: src/Model.js (hunk 3 above)
Comment (brenolf) [inline src/Model.js]: Hmm Mongo's stuff. Is this reusable if we manage to add support to SQL dbs? i.e. Is this Model logic or *Mongo's* Model logic?

Comment (mccraveiro) [inline src/Model.js]: I think its better to focus on mongo 100% right now. However I think id should be present in every document as ActiveRecord do even though its not an Mongo Document.

File: src/Cheddar.js (hunk 2 above)
Comment (brenolf) [inline src/Cheddar.js]: **Edit:** Don't test it, at least for now. I completely forgot [about this](http://en.wikipedia.org/wiki/Cyclomatic_complexity#Implications_for_software_testing). It's cyclomatic index is one, no need to test at all *for now*.

File: src/Model.js (hunk 3 above)
Comment (brenolf) [inline src/Model.js]: Did I get it wrong or this id is model-related? If so, ActiveRecord::Base.connection objects do nothing about setting ids for the "models"; ActiveRecord just eases the way the models handle with their own tables' PKs. I'm afraid things get complicated if we focus on mongo right away :worried:

Comment (mccraveiro) [inline src/Model.js]: With mongodb you can create IDs (PKs) before inserting the document on the DB :smiley:

**Pull request merged**